pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix

```diff
@@ -9,13 +9,17 @@
 , openssl
 , pkg-config
 , postgresql
+, rustPlatform
 }:
 
 let
   # Upstream only works with clang 16, so we're pinning it here to
   # avoid future incompatibility.
   # See https://docs.pgvecto.rs/developers/development.html#environment, step 4
   clang = clang_16;
+  rustPlatform' = rustPlatform // {
+    bindgenHook = rustPlatform.bindgenHook.override { inherit clang; };
+  };
```

Comment:
I'd prefer a full `makeRustPlatform` call.

Reply:
I tried that, but it seems to not be overriding as expected.
```
nix-repl> (pkgs.makeRustPlatform {rustc = pkgs.rustc; cargo = pkgs.cargo; clang = pkgs.clang_17;}).bindgenHook.clang
«derivation /nix/store/j16048gsic45ijdpw82172pfxdz9nw2w-clang-wrapper-16.0.6.drv»
```

Am I doing it wrong?

Reply:
Ah, `makeRustPlatform` doesn't take a `clang` arg but does accept arbitrary args, so it doesn't throw an error to do this.

I don't see another option short of refactoring `makeRustPlatform` then.

I really don't know how this all works together; doesn't rust internally depend on clang aswell? Shouldnt'y you need to override that aswell?

I'd like input of the rust people on this. It appears they were pinged already.

Reply:
This is what we do for firefox: https://github.com/NixOS/nixpkgs/blob/9d422aeafc8780e1efe49f5891b70685d9c05c40/pkgs/applications/networking/browsers/firefox/common.nix#L177

Reply:
Not sure if it also applies to this.

Reply:
@Mic92 Thanks for you help, but it seems that it still doesn't work. Let me know if I'm using this wrong:
```
nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.overrideCC pkgs.llvmPackages.stdenv (llvmPackages.stdenv.override { cc = pkgs.clang_17; }); }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»

nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.overrideCC pkgs.llvmPackages.stdenv.override { cc = pkgs.clang_17; }; }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»

nix-repl> (pkgs.makeRustPlatform { inherit (pkgs) rustc cargo; stdenv = pkgs.llvmPackages.stdenv.override { cc = pkgs.clang_17; }; }).bindgenHook.clang
«derivation /nix/store/c9ym6jk1pf0bx62j29g51k00pp16zf9l-clang-wrapper-16.0.6.drv»
```

For context, this is what I want to achieve (this is the current solution in the code):
```
nix-repl> (pkgs.rustPlatform.bindgenHook.override { clang = pkgs.clang_17; }).clang
«derivation /nix/store/cxxk9njrx1zm5ynmsi0wyy9dv58h2zci-clang-wrapper-17.0.6.drv»
```

(Also, I do want clang 16, but I want to pin it; I'm just using clang 17 in the repl to make sure it is being overridden)

Is there something I'm overlooking?